src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -783,18 +783,36 @@ public function prepare_item_for_response( $item, $request ) {
 					}
 
 					$size_data['source_url'] = $image_src[0];
+
+					if ( empty( $size_data['sources'] ) || ! is_array( $size_data['sources'] ) ) {
+						continue;
+					}
+
+					$directory = dirname( $size_data['source_url'] );
+					foreach ( $size_data['sources'] as $mime => &$mime_details ) {
```

Comment:
here you have used `&$mime_details`, any specific reason to use referenced variable?

Reply:
Please check Felix's feedback here https://github.com/WordPress/wordpress-develop/pull/3025#discussion_r930032070 for more information.

Reply:
@mehulkaklotar This is needed to actually modify the value directly. Otherwise in a `foreach` loop the assignment below won't have an effect.